File: android/src/main/java/com/shovelapp/devfellow/mapboxnavi/MapboxNavigationPluginPlugin.java
package com.shovelapp.devfellow.mapboxnavi;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "MapboxNavigationPlugin")
public class MapboxNavigationPluginPlugin extends Plugin {

    private MapboxNavigationPlugin implementation = new MapboxNavigationPlugin();

    @PluginMethod
    public void echo(PluginCall call) {
        String value = call.getString("value");

        JSObject ret = new JSObject();
        ret.put("value", implementation.echo(value));
        call.resolve(ret);
    }


    @PluginMethod void navigateToDestination(PluginCall call){
      String userId = call.getString("userId");
      double latitude = call.getDouble("latitude");
      double longitude = call.getDouble("longitude");

      implementation.navigateToDestination(latitude, longitude, userId);

      call.resolve();
    }
}

File: ios/Plugin/MapboxNavigationPluginPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(MapboxNavigationPluginPlugin)
public class MapboxNavigationPluginPlugin: CAPPlugin {
    private let implementation = MapboxNavigationPlugin()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }


    @objc func navigateToDestination(_ call: CAPPluginCall) {
        guard let latitude = Double(call.getString("latitude") ?? "0") else { return   }
        guard let longitude = Double(call.getString("longitude") ?? "0") else { return  }
      let userId = call.getString("userId") ?? ""

        implementation.navigateToDestination(latitude,longitude: longitude, userId: userId)

      call.resolve()


    }
}
